speechbrain/core.py

```diff
@@ -1078,36 +1096,41 @@ def fit(
                             self._save_intra_epoch_ckpt()
                         last_ckpt_time = time.time()
 
-            # Run train "on_stage_end" on all processes
-            self.on_stage_end(Stage.TRAIN, self.avg_train_loss, epoch)
-            self.avg_train_loss = 0.0
-            self.step = 0
-
-            # Validation stage
-            if valid_set is not None:
-                self.on_stage_start(Stage.VALID, epoch)
-                self.modules.eval()
-                avg_valid_loss = 0.0
-                with torch.no_grad():
-                    for batch in tqdm(
-                        valid_set, dynamic_ncols=True, disable=not enable
-                    ):
-                        self.step += 1
-                        loss = self.evaluate_batch(batch, stage=Stage.VALID)
-                        avg_valid_loss = self.update_average(
-                            loss, avg_valid_loss
+                # Run train "on_stage_end" on all processes
+                self.on_stage_end(Stage.TRAIN, self.avg_train_loss, epoch)
+                self.avg_train_loss = 0.0
+                self.step = 0
+
+                # Validation stage
+                if valid_set is not None:
+                    self.on_stage_start(Stage.VALID, epoch)
+                    self.modules.eval()
+                    avg_valid_loss = 0.0
+                    with torch.no_grad():
+                        for batch in tqdm(
+                            valid_set, dynamic_ncols=True, disable=not enable
+                        ):
+                            self.step += 1
+                            loss = self.evaluate_batch(batch, stage=Stage.VALID)
+                            avg_valid_loss = self.update_average(
+                                loss, avg_valid_loss
+                            )
+
+                            # Debug mode only runs a few batches
+                            if self.debug and self.step == self.debug_batches:
+                                break
+
+                        # Only run validation "on_stage_end" on main process
+                        self.step = 0
+                        run_on_main(
+                            self.on_stage_end,
+                            args=[Stage.VALID, avg_valid_loss, epoch],
                         )
 
-                        # Debug mode only runs a few batches
-                        if self.debug and self.step == self.debug_batches:
-                            break
-
-                    # Only run validation "on_stage_end" on main process
-                    self.step = 0
-                    run_on_main(
-                        self.on_stage_end,
-                        args=[Stage.VALID, avg_valid_loss, epoch],
-                    )
+            train_time = time.time() - last_ckpt_time
```

Comment:
This feature shouldn't be part of the PR, please remove :-) We can discuss it elsewhere, but it requires to consult the whole team.

Reply:
will do